clang_tidy/run_clang_tidy.sh

```diff
@@ -0,0 +1,27 @@
+#! /bin/bash
+# Usage: run_clang_tidy <CONFIG> <OUTPUT> [ARGS...]
+set -ue
+
+CLANG_TIDY_BIN=$1
+shift
+
+CONFIG=$1
+shift
+
+OUTPUT=$1
+shift
+
+# .clang-tidy config file has to be placed in the current working directory
```

Comment:
This might be the piece that was missing in the `clang_format` implementation that was preventing it from working unless the project provided the `.clang_format` file.